File: frontend-next/lib/chat/assistant-context.ts
import type { AssistantDataSnapshot } from "./assistant-types"

/** LLM 시스템 프롬프트용 데이터 스냅샷 텍스트 */
export function formatAssistantDataContext(snapshot: AssistantDataSnapshot): string {
  const { performance, dashboard, kanban, organization, ebooks, surveys } =
    snapshot

  const monthLines = Object.entries(performance.byMonth)
    .filter(
      ([, m]) =>
        m.planBudget > 0 ||
        m.actualExpense > 0 ||
        m.planPeople > 0 ||
        m.actualPeople > 0,
    )
    .map(
      ([month, m]) =>
        `  ${month}: 계획 예산 ${m.planBudget.toLocaleString()}원 / 실적 지출 ${m.actualExpense.toLocaleString()}원 (인원 계획 ${m.planPeople}·실적 ${m.actualPeople})`,
    )

  const subProjectLines = performance.subProjects.map((name) => {
    const m = performance.bySubProject[name]
    if (!m) return null
    return `  ${name}: 계획 예산 ${m.planBudget.toLocaleString()}원 / 실적 ${m.actualExpense.toLocaleString()}원`
  })

  return [
    `데이터 기준 시각: ${snapshot.generatedAt}`,
    "",
    "## 대시보드",
    ...dashboard.stats.map(
      (s) => `- ${s.label}: ${s.value}${s.unit} (${s.description})`,
    ),
    ...dashboard.progress.map((p) => `- ${p.label}: ${p.value}%`),
    "",
    "## 계획/실적 입력 (행 ${performance.rowCount}건)",
    `- 전체 계획 예산: ${performance.totals.planBudget.toLocaleString()}원`,
    `- 전체 실적 지출: ${performance.totals.actualExpense.toLocaleString()}원`,
    `- 전체 계획 인원/횟수: ${performance.totals.planPeople}명 / ${performance.totals.planCount}회`,
    `- 전체 실적 인원/횟수: ${performance.totals.actualPeople}명 / ${performance.totals.actualCount}회`,
    "- 월별:",
    ...monthLines,
    "- 세목(세부사업명)별:",
    ...subProjectLines.filter(Boolean),
    "",
    "## 칸반",
    `- 프로젝트 ${kanban.projectCount}개, 업무 카드 ${kanban.taskCount}건`,
    ...Object.entries(kanban.tasksByStatus).map(
      ([status, count]) => `  - ${status}: ${count}건`,
    ),
    "",
    "## 조직",
    `- 부서 ${organization.departmentCount}개, 직원 ${organization.employeeCount}명`,
    "",
    "## 전자책",
    `- 자료 ${ebooks.bookCount}권, 카테고리: ${ebooks.categories.join(", ")}`,
    "",
    "## 설문",
    `- 설문 ${surveys.totalCount}개: ${surveys.titles.join(", ")}`,
  ].join("\n")
}

File: backend/app/modules/chat/assistant_context.py
"""Format assistant snapshot for LLM system prompt."""

from __future__ import annotations

from typing import Any


def format_assistant_data_context(snapshot: dict[str, Any]) -> str:
    performance = snapshot["performance"]
    dashboard = snapshot["dashboard"]
    kanban = snapshot["kanban"]
    organization = snapshot["organization"]
    ebooks = snapshot["ebooks"]
    surveys = snapshot["surveys"]

    month_lines = []
    for month, metrics in performance.get("byMonth", {}).items():
        if (
            metrics.get("planBudget", 0) > 0
            or metrics.get("actualExpense", 0) > 0
            or metrics.get("planPeople", 0) > 0
            or metrics.get("actualPeople", 0) > 0
        ):
            month_lines.append(
                f"  {month}: 계획 예산 {metrics['planBudget']:,}원 / "
                f"실적 지출 {metrics['actualExpense']:,}원 "
                f"(인원 계획 {metrics['planPeople']}·실적 {metrics['actualPeople']})"
            )

    sub_project_lines = []
    for name in performance.get("subProjects", []):
        metrics = performance.get("bySubProject", {}).get(name)
        if metrics:
            sub_project_lines.append(
                f"  {name}: 계획 예산 {metrics['planBudget']:,}원 / "
                f"실적 지출 {metrics['actualExpense']:,}원"
            )

    totals = performance.get("totals", {})
    stat_lines = [
        f"- {s['label']}: {s['value']}{s['unit']} ({s['description']})"
        for s in dashboard.get("stats", [])
    ]
    progress_lines = [f"- {p['label']}: {p['value']}%" for p in dashboard.get("progress", [])]

    return "\n".join(
        [
            f"데이터 기준 시각: {snapshot['generatedAt']}",
            "",
            "## 대시보드",
            *stat_lines,
            *progress_lines,
            "",
            f"## 계획/실적 입력 (행 {performance.get('rowCount', 0)}건)",
            f"- 전체 계획 예산: {totals.get('planBudget', 0):,}원",
            f"- 전체 실적 지출: {totals.get('actualExpense', 0):,}원",
            f"- 전체 계획 인원/횟수: {totals.get('planPeople', 0)}명 / {totals.get('planCount', 0)}회",
            f"- 전체 실적 인원/횟수: {totals.get('actualPeople', 0)}명 / {totals.get('actualCount', 0)}회",
            "- 월별:",
            *month_lines,
            "- 세목(세부사업명)별:",
            *sub_project_lines,
            "",
            "## 칸반",
            f"- 프로젝트 {kanban.get('projectCount', 0)}개, 업무 카드 {kanban.get('taskCount', 0)}건",
            *[
                f"  - {status}: {count}건"
                for status, count in kanban.get("tasksByStatus", {}).items()
            ],
            "",
            "## 조직",
            f"- 부서 {organization.get('departmentCount', 0)}개, 직원 {organization.get('employeeCount', 0)}명",
            "",
            "## 전자책",
            f"- 자료 {ebooks.get('bookCount', 0)}권, 카테고리: {', '.join(ebooks.get('categories', []))}",
            "",
            "## 설문",
            f"- 설문 {surveys.get('totalCount', 0)}개: {', '.join(surveys.get('titles', []))}",
        ]
    )
